lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -661,6 +664,27 @@ impl<TGraphLinks: GraphLinks> VectorIndex for HNSWIndex<TGraphLinks> {
         let pool = rayon::ThreadPoolBuilder::new()
             .thread_name(|idx| format!("hnsw-build-{idx}"))
             .num_threads(max_rayon_threads(self.config.max_indexing_threads))
+            .spawn_handler(|thread| {
+                let mut b = thread::Builder::new();
+                if let Some(name) = thread.name() {
```

Comment:
Did you check if the thread name shows up in htop?

Reply:
Yes: 
![image](https://github.com/qdrant/qdrant/assets/856222/22cea0ed-828f-4f59-a9b6-96f31fb883df)

I've sourced the above code from [here](https://docs.rs/rayon/1.8.0/rayon/struct.ThreadPoolBuilder.html#examples-1), but have added the call for setting the thread priority.